osu.Game/Online/Notifications/NotificationsClient.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Online.Chat;
+
+namespace osu.Game.Online.Notifications
+{
+    /// <summary>
+    /// An abstract client which receives notification-related events (chat/notifications).
+    /// </summary>
+    public abstract class NotificationsClient : PersistentEndpointClient
+    {
+        public Action<Channel>? ChannelJoined;
+        public Action<List<Message>>? NewMessages;
+        public Action? PresenceReceived;
+
+        protected readonly IAPIProvider API;
+
+        private bool enableChat;
+        private long lastMessageId;
+
+        protected NotificationsClient(IAPIProvider api)
+        {
+            API = api;
+        }
+
+        public bool EnableChat
```

Comment:
I doubt this is ever going to be a thing, but setting this to false doesn't seem to do anything.

Also curious why this is a `bool` rather than a `StartChat` method like the invocation.